### PR TITLE
[KYUUBI #6496] Step 1/2: Port ZOOKEEPER-4377 to Kyuubi shaded ZooKeeper 3.4 and 3.6

### PR DESCRIPTION
**Pull request by Madhukar525722**
### _Why are the changes needed?_

This PR is to resolve the [Kyuubi #6496](https://github.com/apache/kyuubi/issues/6496) issue raised. The change is split into two steps
1. First is to overwrite the KeeperException class, raised in this PR (KeeperException for zookeeper34 is pulled from 
[zookeeper-3.4.14](https://github.com/apache/zookeeper/blob/release-3.4.14/zookeeper-server/src/main/java/org/apache/zookeeper/KeeperException.java) for zookeeper36 is pulled from [zookeeper-3.6](https://github.com/apache/zookeeper/blob/branch-3.6/zookeeper-server/src/main/java/org/apache/zookeeper/KeeperException.java) )
2. Step 2 will be for backporting the ZOOKEEPER-4377 change

### _How was this patch tested?
This was built locally and tested in local using following steps:
1. perform `build/mvn clean package` to generate `kyuubi-relocated-zookeeper-parent/kyuubi-relocated-zookeeper-34/target/kyuubi-relocated-zookeeper-34-0.5.0-SNAPSHOT.jar`
2. download and install [apache-kyuubi-1.9.1-bin.tgz](https://dlcdn.apache.org/kyuubi/kyuubi-1.9.1/apache-kyuubi-1.9.1-bin.tgz)
3. replace `KYUUBI_HOME/jars/kyuubi-relocated-zookeeper-34-0.3.0-SNAPSHOT.jar` with `kyuubi-relocated-zookeeper-34-0.5.0-SNAPSHOT.jar`
4. perform `bin/kyuubi-zk-cli -server xxxxx:2181` to connect a Zookeeper cluster

### CODE REVIEW

**Comment by pan3793** [comment]
@Madhukar525722 good job, could you please spilt it into 2 PRs? So that we can keep the clear git commit history.

**Comment by Madhukar525722** [comment]
Updated this pull request, consisting step 1 only

**Comment by Madhukar525722** [comment]
@pan3793 Please review the change

File: kyuubi-relocated-hive-metastore-client/pom.xml [hunk 1]
```diff
@@ -36,6 +36,7 @@ under the License.
         <hadoop.version>3.3.6</hadoop.version>
         <httpclient.version>4.5.14</httpclient.version>
         <httpcore.version>4.4.16</httpcore.version>
+        <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
```
Comment (pan3793) [inline kyuubi-relocated-hive-metastore-client/pom.xml:39]: change is not related.

note: this repo is supposed to be built and published via Java 8, not need to handle Java 11 or later cases

**Review by pan3793 — APPROVED** [review]
LGTM, only a nit

File: kyuubi-relocated-hive-metastore-client/pom.xml (hunk 1 above)
Comment (Madhukar525722) [reply kyuubi-relocated-hive-metastore-client/pom.xml]: Ohh, I didnt observed and used Java17, it worked fine with Java8. And I have taken care of this. Thanks for the review @pan3793

**Comment by pan3793** [comment]
Thanks, merged to master(0.5.0)/0.4.1

**Pull request closed without merging**